crypt.go

```diff
@@ -135,6 +139,23 @@ func activate(volumeName, sourceDevicePath string, key []byte, options []string)
 		}
 		done <- true
 	}()
+
+	f, err := os.OpenFile(fifoPath, os.O_WRONLY, 0)
+	if err != nil {
+		// If we fail to open the write end, the read end will be blocked in open()
+		cmd.Process.Kill()
+		return xerrors.Errorf("cannot open FIFO for passing key to systemd-cryptsetup: %w", err)
+	}
+
+	if _, err := f.Write(key); err != nil {
+		f.Close()
+		// The read end is open and blocked inside read(). Closing our write end will result in the
+		// read end returning 0 bytes (EOF) and exitting cleanly.
+		cmd.Wait()
+		return xerrors.Errorf("cannot pass key to systemd-cryptsetup: %w", err)
+	}
+
+	f.Close()
```

Comment:
Should we check if this Close() call fails?

Reply:
I did think about this, but I'm not sure what to do with the error. Perhaps just killing cryptsetup, as it will hang if the close fails anyway?

Reply:
Yes, I think it's the only thing we can do.

Reply:
I've merged it as-is for now. I had another think about it and I'm not really sure what would be the appropriate thing to do with the error. As close() does no I/O, there's nothing that could cause the syscall to fail in the kernel (it can only fail if the fd is invalid or already closed). As long as the close syscall executes, the cryptsetup process is guaranteed to continue and perform the requested action. I need to have a look at the go code to see if os.File.Close() can fail for any reason on the go side, and whether it can fail after the syscall completes successfully - if it can, then that would mean that both AddRecoveryKeyToLUKS2Container and ChangeLUKS2KeyUsingRecoveryKey could both return an error in some circumstances whilst still successfully making changes to the LUKS header.